src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -74,22 +73,36 @@ private void processZipFile(CloudBlobContainer container, String zipFilename)
         log.info("Processing zip file {}", zipFilename);
 
         CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(zipFilename);
-        BlobInputStream blobInputStream = cloudBlockBlob.openInputStream();
-
-        //Zip file will include metadata.json and collection of pdf documents
-        try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
-            ZipFileProcessor zipFileProcessor = processZipInputStream(zis, zipFilename, container.getName());
-
-            if (zipFileProcessor != null) {
-                processParsedEnvelopeDocuments(
-                    zipFileProcessor.getEnvelope(),
-                    zipFileProcessor.getPdfs(),
-                    cloudBlockBlob
-                );
+
+        CloudBlockBlob blobWithLeaseAcquired = acquireLease(cloudBlockBlob, container.getName(), zipFilename);
+
+        if (Objects.nonNull(blobWithLeaseAcquired)) {
+            BlobInputStream blobInputStream = blobWithLeaseAcquired.openInputStream();
+
+            // Zip file will include metadata.json and collection of pdf documents
+            try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
+                ZipFileProcessor zipFileProcessor = processZipInputStream(zis, zipFilename, container.getName());
+
+                if (zipFileProcessor != null) {
+                    processParsedEnvelopeDocuments(
+                        zipFileProcessor.getEnvelope(),
+                        zipFileProcessor.getPdfs(),
+                        blobWithLeaseAcquired
+                    );
+                }
             }
         }
     }
 
+    private CloudBlockBlob acquireLease(CloudBlockBlob cloudBlockBlob, String containerName, String zipFilename) {
+        // TODO - Lease time  needs to be made configurable
+        return errorWrapper.wrapAcquireLeaseFailure(containerName, zipFilename, () -> {
+            // The value of the parameter 'leaseTimeInSeconds' should be between 15 and 60 seconds.
+            cloudBlockBlob.acquireLease(15, null);
```

Comment:
What happens if for some reason (such as high load) processing takes longer than 15secs? Does another thread acquire the lease and processes the file? Have we got a test for this?

Reply:
Yes after 15 secs another thread can acquire the lease. Waiting for 15 secs in a test would be too long imo. We can rely on Azure storage for this and need not test that.

Reply:
So if processing takes longer than 15secs we could have a situation in which a blob is processed twice? That is:
- T1 acquires lease and starts (long) processing.
- Lease expires
- T2 acquires lease and starts processing.
 Or is there a way to prevent that?

Reply:
Single Blob processing should not take too much time(15 secs is too much) and if we expect and we know that if might take too long upfront then we should configure lease timeout > 15 secs or keep renewing lease. Unless proven that it takes longer than 15 secs I would not complicate it at this point as we need to know when to renew lease and where it is taking time.

Reply:
The problem I see is that, even if most of the time it will take less than 15secs, occasionally (e.g.  because of high load) it could take more than 15secs. We should have something in place that prevents the lease to be acquired again in those cases.

Reply:
Like I said earlier unless we know where it takes time for example it might take time for doc upload then we should fix that rather than fixing lease. Feel free to raise a ticket as this is beyond the scope of this ticket.